### PR TITLE
Prevent file downloads if the disk would be full

### LINKED ISSUE #6697: Kolibri does not report to the user that disk is full during channel import.
### Observed behavior
When importing a channel from Kolibri Studio (online)/Internet (example: Khan Academy English), Kolibri does not report to the user if the disk is full.  This causes the import to "freeze", which eventually makes Kolibri unresponsive.

On a similar note, Kolibri also does not prompt the user that the size of the channel to import is larger than the available disk space.

#### Screenshots

I started the import the day before and when I came back to the page the following morning: Kolibri "froze" at 80% during my import.
Screenshot 2020-03-25 11 32 33

After I reloaded the page many times, I saw this.
Screenshot 2020-03-25 13 34 56

Today, I re-created a VM with a 20GB disk and tried to import channels with total size greater than the available disk space but Kolibri did not prompt me of an issue.
Screenshot 2020-03-26 11 52 28

Tried to import channels with 20GB+ in size twice to make sure but Kolibri still went on with the import.
Screenshot 2020-03-26 11 53 29

### Expected behavior
User must be notified when the disk is (almost) full during a channel import and then stop the import process.  Kolibri then, must still be working and shouldn't crash.

Another expectation is to notify the user that the channel to be imported is too big for the available/free disk space, to prevent Kolibri crashes like this.

### User-facing consequences
Kolibri stops working when the import process is interrupted because the disk is already full.

### Errors and logs

From `~/.kolibri/logs/kolibri.txt`:
```
...
ERROR 2020-03-24 03:23:37,985 importcontent An error occurred during content import: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
INFO 2020-03-24 03:23:38,801 importcontent Waiting for 30 seconds before retrying import: https://studio.learningequality.org/content/storage/9/9/99441f1d6debd2a8870a845e22de7525.vtt
ERROR 2020-03-24 12:58:58,202 importcontent An error occurred during content import: [Errno 28] No space left on device
INFO 2020-03-24 12:58:58,568 annotation Setting availability to True of 55222 LocalFile objects based on passed in checksums
ERROR 2020-03-24 12:59:00,639 worker Job a29c1ebc46c3452f9b9d61cc3aaf82a4 raised an exception: Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/base.py", line 724, in _commit_impl
    self.engine.dialect.do_commit(self.connection)
  File "/usr/lib/python3/dist-packages/kolibri/dist/sqlalchemy/engine/default.py", line 462, in do_commit
    dbapi_connection.commit()
sqlite3.OperationalError: database or disk is full
...
```

### Steps to reproduce

1. Create an Ubuntu Server 16.04.6 Virtual Machine with a 20GB disk space.
1. Download the `.deb` Ubuntu installer for [Kolibri v0.13.2-beta1](https://github.com/learningequality/kolibri/releases/tag/v0.13.2-beta1) and install following the [instructions here](https://kolibri.readthedocs.io/en/latest/install/ubuntu-debian.html#install-from-a-deb-file).
1. Connect to the VM using a web browser and setup Kolibri.  Example: http://192.168.1.10:8080/
1. Import Khan Academy English channel content via Kolibri Studio (online).  Select content with total size larger than the VM disk free space (>= 19GB).  Example: `Math` and `Science` nodes currently total to 19GB.

**Notes**

1. Content size of 19GB is okay because Ubuntu Server 16.04.6 and Kolibri uses about 2GB of data, which leaves about 18GB of free space in the 20GB VM disk.
1. For faster testing, use a smaller disk size for your VM like 4GB.  Just make sure that the channel size exceeds the free space of roughly >= 2GB.

### Context
 * Kolibri version: [0.13.2-beta1](https://github.com/learningequality/kolibri/releases/tag/v0.13.2-beta1)
 * Operating system: Ubuntu 16.04.6 LTS on a Virtual Machine
 * Browser: Safari v13.0.5 (web browser used to access Kolibri)

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Adds logic from the bulk channel import page into the granular content selection page
* Displays current free disk space in the granular content selection page
* Makes import content tasks fail if there is insufficient free disk space
* Adds a 'minimum free disk space' option to ensure that we leave a configurable buffer
* Sets the default to 250MB

## References
Fixes #6697 but with a slight deviation from the design to allow for direct string reuse.

## Reviewer guidance
Does Kolibri prevent you from downloading files that would fill up your drive?
Does Kolibri error a task if the drive fills up while downloading? (download a large file independently to fill up the space).

To help with testing this, you can use the new option:
```
KOLIBRI_MINIMUM_DISK_SPACE=63000000000
```
Will reserve 63GB of space on your disk that Kolibri will not attempt to use.

Animation of new interaction:
freespace

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [x] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/utils/options.py [hunk 1]
```diff
@@ -498,6 +498,14 @@ def url_prefix(value):
...
             """,
         },
+        "MINIMUM_DISK_SPACE": {
+            "type": "integer",
+            "default": 250000000,
+            "description": """
+                The minimum free disk space in bytes that Kolibri should try to maintain on the device. This will
+                be used as the floor value to prevent Kolibri completely filling the disk during file import.
+            """,
+        },
```
Comment (indirectlylit) [inline kolibri/utils/options.py]: could we define this in MB (or maybe a decimal GB) to be a bit more user-friendly?

File: kolibri/plugins/device/assets/src/views/SelectContentPage/index.vue [hunk 2]
```diff
@@ -46,6 +47,13 @@
         >
           {{ $tr('problemTransferringContents') }}
         </UiAlert>
+        <UiAlert
+          v-show="transferFileSize > availableSpace"
+          :dismissible="false"
+          type="error"
+        >
+          {{ spaceTranslator.$tr('notEnoughSpaceForChannelsWarning') }}
+        </UiAlert>
```
Comment (indirectlylit) [inline kolibri/plugins/device/assets/src/views/SelectContentPage/index.vue:56]: filed a follow-up issue regarding in-page errors: https://github.com/learningequality/kolibri/issues/8690

**Review by indirectlylit — COMMENTED** [review]
This seems to be mostly working. However, the minimum space requirement seems to only be used in the 'resource selection' import mode, not the 'channel selection' import mode.

Non-blocking issue: the error doesn't show up until something has been selected. I would expect it to show up immediately:

**Comment by indirectlylit** [comment]
cc @radinamatic regarding user docs and gherkins

File: kolibri/utils/options.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/utils/options.py]: My main concern would be the frequent confusion between GiB and GB, and the fact that these may be used differently when people are examining their local disks. Using bytes is unambiguous if somewhat verbose.

**Comment by radinamatic** [comment]
@pcenov We need to add the relevant Gherkin scenario to test this in (I presume) **beta3**.

**Comment by rtibbles** [comment]
Updated with the following:
Screenshot from 2021-11-15 12-19-28

Screenshot from 2021-11-15 12-15-56

Further, the specification of the minimum free space is now able to accept an integer, or a file size demoninated in B, KB, MB, GB, TB, or PB.

File: kolibri/utils/options.py (hunk 1 above)
Comment (indirectlylit) [reply kolibri/utils/options.py]: For what it's worth:

* Android [appears to report decimal](https://android.stackexchange.com/a/129682)
* MacOS [reports decimal](https://support.apple.com/en-us/HT201402)
* Ubuntu [shows decimal in the GUI](https://askubuntu.com/a/860890) and `df` can show either

Windows is the only one that seems to [still report base2](https://support.hp.com/lt-en/document/ish_3046042-2787120-16)

File: kolibri/utils/options.py [hunk 3]
```diff
@@ -62,7 +63,7 @@ def calculate_thread_pool():
     if psutil:
         MIN_MEM = 2
         MAX_MEM = 6
-        total_memory = psutil.virtual_memory().total / pow(2, 30)  # in Gb
+        total_memory = psutil.virtual_memory().total / pow(10, 9)  # in Gb
```
Comment (indirectlylit) [inline kolibri/utils/options.py]: for full pedantic correctness :D

```suggestion
        total_memory = psutil.virtual_memory().total / pow(10, 9)  # in GB
```

File: kolibri/utils/options.py [hunk 4]
```diff
@@ -498,6 +507,15 @@ def url_prefix(value):
                 data that is returned to our telemetry server.
             """,
         },
+        "MINIMUM_DISK_SPACE": {
+            "type": "bytes",
+            "default": "250MB",
+            "description": """
+                The minimum free disk space that Kolibri should try to maintain on the device. This will
+                be used as the floor value to prevent Kolibri completely filling the disk during file import.
+                Value can either be an integer in bytes or suffixed with a unit from B, KB, MB, GB, TB, or PB.
```
Comment (indirectlylit) [inline kolibri/utils/options.py]: It appears that the suffixed versions can be fractional.

```suggestion
                Value can either be a number suffixed with a unit (e.g. MB, GB, TB) or an integer number of bytes.
```

While technically we may allow KB or PB, not sure it's helpful to mention here.

**Review by indirectlylit — APPROVED** [review]
Tested and works great!

Minor suggestion regarding docs.

**Pull request merged**